source/how-tos/boot-acrn.rst

```diff
@@ -0,0 +1,357 @@
+
+.. |UP2_FLASHING| raw:: html
+
+   <a href="https://wiki.up-community.org/BIOS_chip_flashing_on_UP_Squared" target="_blank">BIOS chip flashing</a>
+
+
+For more details on setting up the hardware, see :ref:`up2-board`.
+
+
+Build |SPN| 
+^^^^^^^^^^^^^^^^
+
+Build and stitch a custom |SPN| with fastboot support
+
+**Step 1:** Install tool chains and clone SBL source code from GitHub
+
+  Run::
+
+    sudo apt-get install -y build-essential iasl python uuid-dev nasm openssl gcc-multilib qemu
+    git clone git@github.com:slimbootloader/slimbootloader.git
+    cd slimbootloader
+    git config --global user.email "email@example.com"
+    git config --global user.name "your name"
+
+  Compile |SPN| just to verify your setup is complete::
+
+    python BuildLoader.py build apl
+
+
+**Step 2:** Download additional required images and tools 
+
+1. Download fastboot executable ``fastboot.elf`` from |FB| to |SPN| source directory
+
+.. |FB| raw:: html
+
+   <a href="https://github.com/intel/kernelflinger/raw/master/prebuilt/board/APL_UP2/fastboot.elf" target="_blank">here</a>
+
+2. Download |UP2| BIOS from |BIOS_V4| and unzip as is in |SPN| source directory
+
+.. |BIOS_V4| raw:: html
+
+   <a href="https://downloads.up-community.org/download/up-squared-uefi-bios-v4-0/" target="_blank">here</a>
+
+3. Download IAS script ``iasimage`` from |IAS| to |SPN| source directory
+
+.. |IAS| raw:: html
+
+   <a href="https://raw.githubusercontent.com/intel/iasimage/master/iasimage" target="_blank">here</a>
+
+   Run as root the following commands::
+
+     # apt-get install python3 python3-pip
+     # pip3 install cryptography==2.2.2
+
+**Step 3:** Generate IAS image
+
+  Run::
+
+    mkdir -p Platform/ApollolakeBoardPkg/SpiIasBin
+    touch cmdline.txt
+    chmod 755 ./iasimage
+    ./iasimage create -o Platform/ApollolakeBoardPkg/SpiIasBin/iasimage1.bin -d ./BootloaderCorePkg/Tools/Keys/TestSigningPrivateKey.pem -i 0x40000 cmdline.txt fastboot.elf
+
+``iasimage1.bin``
+  The binary file will be added into |SPN| image during build process
+
+**Step 4:** Build and stitch
+
+  Run::
+
+    python BuildLoader.py build apl
+    python Platform/ApollolakeBoardPkg/Script/StitchLoader.py -i UPA1AM40.bin -s Outputs/apl/Stitch_Components.zip -o sbl_up2_ifwi.bin -p 0xAA00000E -b bios.bin
+    python BootloaderCorePkg/Tools/GenCapsuleFirmware.py -k ./BootloaderCorePkg/Tools/Keys/TestSigningPrivateKey.pem -o FwuImage.bin -b bios.bin
+
+``sbl_up2_ifwi.bin``
+  The binary file to be flashed by BIOS flash tool or SPI programmer
+
+``FwuImage.bin``
+  The binary file to be copied to USB key to upgrade SBL from shell command
+
+
+Flash |SPN|
+^^^^^^^^^^^^^^^^
+
+If you are using UEFI BIOS on |UP2|, you will need to update to SBL for the first time.
+
+
+**Step 1:** BIOS flash tool is included in the same BIOS package downloaded |BIOS_V4|
+
+BIOS flash tool is named ``Fpt_xxx.efi`` inside the BIOS package.

```

Comment:
Are we able to provide a path or must the user `find` it?

Reply:
What I mean is: is this file in the root of the "BIOS package" or is it buried in some hierarchy of directories?

Reply:
All files inside the BIOS zip file downloaded from UP2 website is in the top level directory.